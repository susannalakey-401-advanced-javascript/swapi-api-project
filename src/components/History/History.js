import React from 'react';
import './History.scss'
import { Link } from 'react-router-dom'




const getUrlFromStorage = () => {
  const urlsRaw = localStorage.getItem('urls');
  if (urlsRaw) {
    return JSON.parse(urlsRaw);
  }
  // if there's nothing in localStorage, send back an empty array
  return [];
}


const History = () => {
  return (
    <div className="search-list">
      <p>Search History</p>
      <ul>
        {getUrlFromStorage().map(url => (
          <li key={url}><Link to={{ pathname: '/', search: `?url=${url}` }}>{url}</Link></li>
        ))}
      </ul>
    </div >
  )
}

export default History