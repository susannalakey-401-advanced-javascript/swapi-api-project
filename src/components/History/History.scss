@import url('https://fonts.googleapis.com/css?family=Hind&display=swap');

.search-list{
  font-family: 'Hind', sans-serif;
  background-color: white;
  text-align: center;
  p {
    padding-top: 20px;
    margin-left: 30px;
    font-size: 24px;
  }
  ul {
    margin: 10px;
  }
  li {
    list-style-type: none;
    
    padding: 10px;
  }
}