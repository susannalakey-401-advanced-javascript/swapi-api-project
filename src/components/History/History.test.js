import React from 'react'
import History from './History'
import { shallow } from 'enzyme'

describe('<History />', () => {
  let tree;

  beforeEach(() => {
    tree = shallow(<History />)
  })

  it('renders to the page', () => {
    expect(tree).toMatchSnapshot();
  })
})