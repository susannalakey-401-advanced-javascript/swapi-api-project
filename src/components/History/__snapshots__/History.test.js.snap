// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> renders to the page 1`] = `
<div
  className="search-list"
>
  <p>
    Search History
  </p>
  <ul />
</div>
`;
